pkg/tcpip/stack/stack.go

```diff
@@ -1151,6 +1151,13 @@ func (s *Stack) FindRoute(id tcpip.NICID, localAddr, remoteAddr tcpip.Address, n
 		s.routeMu.RLock()
 		defer s.routeMu.RUnlock()
 
+		// We want to find the route that has the longest prefix length,
+		// so that more specific routes will win.
```

Comment:
should this be in its own change?

Reply:
I'm not sure it makes sense for it to be separate, the goal of this PR is selecting the most accurate route for the destination as per the routing table, regardless of local addresses.